PostHog/Autocapture/PostHogAutocaptureIntegration.swift

```diff
@@ -11,11 +11,35 @@
     private let elementsChainDelimiter = ";"
 
     class PostHogAutocaptureIntegration: AutocaptureEventProcessing {
-        private let postHogConfig: PostHogConfig
+        private static var integrationInstalledLock = NSLock()
+        private static var integrationInstalled = false
+
+        private weak var postHogInstance: PostHogSDK?
```

Comment:
does it have to be `weak` if we control when we pass the instance and we set it to null?

Reply:
nope, but I don't know how the swift reference counting works deeply, so as long as this instance isn't null randomly, all good